kaolin/datasets/shrec.py

```diff
@@ -34,71 +36,125 @@ class SHREC16(Dataset):
         root (str): Path to the root directory of the dataset.
         categories (list): List of categories to load (each class is
             specified as a string, and must be a valid `SHREC16`
-            category).
-        mode (str, choices=['train', 'test']): Whether to load the
-            'train' split or the 'test' split
+            category). If this argument is not specified, all categories
+            are loaded by default.
+        train (bool): If True, return the train split, else return the test
+            split (default: True).
     Returns:
-        dict: Dictionary with keys: 'vertices' : vertices , 'faces' : faces
+        .. code-block::
+
+           dict: {
+                attributes: {path: str, category: str, label: int},
+                data: kaolin.rep.TriangleMesh
+           }
+
+        path: The filepath to the .obj file on disk.
+        category: A human-readable string describing the loaded sample.
+        label: An integer (in the range :math:`[0, \text{len(categories)}]`)
+            and can be used for training classifiers for example.
+        vertices: Vertices of the loaded mesh (:math:`(*, 3)`), where :math:`*`
+            indicates a positive integer.
+        faces: Faces of the loaded mesh (:math:`(*, 3)`), where :math:`*`
+            indicates a positive integer.
+
+    Example:
+        >>> dataset = SHREC16(root='/path/to/SHREC16/', categories=['alien', 'ants'], train=False)
+        >>> sample = dataset[0]
+        >>> sample["attributes"]["path"]
+        /path/to/SHREC16/alien/test/T411.obj
+        >>> sample["attributes"]["category"]
+        alien
+        >>> sample["attributes"]["label"]
+        0
+        >>> sample["data"].vertices.shape
+        torch.Size([252, 3])
+        >>> sample["data"].faces.shape
+        torch.Size([500, 3])
 
     """
 
-    def __init__(self, root: str, categories: list = ['alien'],
-                 mode: list = 'train'):
-
-        super(SHREC16, self).__init__()
-
-        if mode not in ['train', 'test']:
-            raise ValueError('Argument \'mode\' must be one of \'train\''
-                             'or \'test\'. Got {0} instead.'.format(mode))
+    def initialize(
+        self,
+        root: str,
+        categories: Iterable = None,
+        train: bool = True,
+    ):
 
         VALID_CATEGORIES = [
```

Comment:
I think this should be static (defined inside the class outside of the function)

Reply:
Yes